src/parser/rogue/outlaw/modules/spells/BladeFlurry.ts

```diff
@@ -0,0 +1,56 @@
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Abilities from 'parser/core/modules/Abilities';
+import SpellUsable from 'parser/shared/modules/SpellUsable';
+import SPELLS from 'common/SPELLS';
+import Events, { CastEvent } from 'parser/core/Events';
+import RESOURCE_TYPES, { getResource } from 'game/RESOURCE_TYPES';
+
+class BladeFlurry extends Analyzer {
+  static dependencies = {
+    abilities: Abilities,
+    spellUsable: SpellUsable,
+  };
+
+  hasTrueBearing: boolean = false;
+  protected abilities!: Abilities;
+  protected spellUsable!: SpellUsable;
+
+  constructor(options: any) {
+    super(options);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell([SPELLS.DISPATCH, SPELLS.EVISCERATE, SPELLS.KIDNEY_SHOT, SPELLS.BETWEEN_THE_EYES, SPELLS.SLICE_AND_DICE]), this.onFinishMove);
+    this.addEventListener(Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.TRUE_BEARING), this.onApplyTrueBearing);
+    this.addEventListener(Events.removebuff.by(SELECTED_PLAYER).spell(SPELLS.TRUE_BEARING), this.onRemoveTrueBearing);
+  }
+
+  onRemoveTrueBearing() {
+    this.hasTrueBearing = false;
+  }
+
+  onApplyTrueBearing() {
+    this.hasTrueBearing = true;
+  }
+
+  onFinishMove(event: CastEvent) {
```

Comment:
So, the nested if's on this are a bit much and half of them probably arent needed.

Like if this.spellUsable.isOnCooldown returns true, then surely cooldownRemaining should also return true, so checking for both of them seems like it would be a waste

Additionally, if the resource isnt combo points, that seems like something you would just return early on and not even get that far to begin with .... but also the finishers you are looking for the event listener, would any of them ever not have combo points as a resource?

```
let extraCDR = 0;
if (this.hasTrueBearing) {
    extraCDR = cpCost * 1000;
}
```
feels like you can just do `extraCDR = this.hasTrueBearing ? cpCost * 1000 : 0;`


Just seems like this is way more complicated than it needs to be

Reply:
Finishing moves can cost energy OR combo points. I'll take a second look at the if statements and attempt to clean it up.

Reply:
Then you can return early and save a nested if ... unless you do something with events that dont have combo points as a resource

```
onFinishMove(event: CastEvent) {
if (!getResource(event.classResources, RESOURCE_TYPES.COMBO_POINTS.id) {
  return;
}
the rest of the stuff
```